src/parsemultipart.jl

```diff
@@ -0,0 +1,269 @@
+line.  The result is returned as an array of tuples.
+
+In the case of a flag the first value of the tuple is false the second value
+is the flag and the third value is nothing.
+
+In the case of a key/value argument the first value is true, the second is the
+key, and the third is the value (or nothing if no value was specified).
+"""
+function content_disposition_extract(str)
+    retval = Vector{Tuple{Bool, SubString, Union{SubString,Nothing}}}()
+    tokens = content_disposition_tokenize(str)
+    total  = length(tokens)
+
+    function strip_quotes(val)
+        if val[1] == '"' && val[end] == '"'
+            SubString(val, 2, lastindex(val) - 1)
+        else
+            val
+        end
+    end
+
+    i = 1
+    while i < total
+        if tokens[i] != ';'
+            pair  = (i + 1 <= total && tokens[i + 1] == "=")
+            key   = strip_quotes(tokens[i])
+            value = (pair && i + 2 <= total && tokens[i + 2] != ";" ? strip_quotes(tokens[i + 2]) : nothing)
+
+            push!(retval, (pair, key, value))
+
+            if pair
+                i += 3
+            else
+                i += 1
+            end
+        else
+            i += 1
+        end
+    end
+    return retval
+end
+
+"""
+    parse_multipart_chunk(chunk)
+
+Parse a single multi-part chunk into a Multipart object.  This will decode
+the header and extract the contents from the byte array.
+"""
+function parse_multipart_chunk(chunk)
+    (startIndex, end_index) = find_header_boundary(chunk)
+
+    headers = String(view(chunk, startIndex:end_index))
+    content = view(chunk, end_index+1:lastindex(chunk))
+
+    disposition = match(r"(?i)Content-Disposition: form-data(.*)\r\n", headers)
+
+    if disposition === nothing
+        @warn "Content disposition is not specified dropping the chunk."
+        return # Specifying content disposition is mandatory
+    end
+
+    name = nothing
+    filename = nothing
+
+    for (pair, key, value) in content_disposition_extract(disposition[1])
+        if pair && key == "name"
+            name = value
+        elseif pair && key == "filename"
+            filename = value
+        end
+    end
+
+    name === nothing && return
+
+    match_contenttype = match(r"(?i)Content-Type: (\S*[^;\s])", headers)
+    contenttype = match_contenttype !== nothing ? match_contenttype[1] : "text/plain" # if content_type is not specified, the default text/plain is assumed
+
+    return Multipart(filename, IOBuffer(content), contenttype, "", name)
+end
+
+"""
+    parse_multipart_body(body, boundary)::Vector{Multipart}
+
+Parse the multipart body received from the client breaking it into the various
+chunks which are returned as an array of Multipart objects.
+"""
+function parse_multipart_body(body::AbstractVector{UInt8}, boundary::AbstractString)::Vector{Multipart}
+    multiparts = Multipart[]
+    idxs = find_multipart_boundaries(body, IOBuffer("--$(boundary)").data)
```

Comment:
I think `.data` is internal and should be accessed with `take!(io)` instead. But that still seems like a backhanded way to get at the data as a byte array. Perhaps nicer just to use

```suggestion
    idxs = find_multipart_boundaries(body, Vector{UInt8}("--"*boundary))
```

Reply:
Fixed this in 67356191.